test/distributed/org/apache/cassandra/distributed/test/sai/IndexAvailabilityTest.java

```diff
@@ -0,0 +1,290 @@
+        for (String ks : keyspaces)
+        {
+            for (String indexName : indexesPerKs)
+            {
+                assertIndexingStatus(cluster, ks, indexName);
+            }
+        }
+    }
+
+    private static void assertIndexingStatus(Cluster cluster, String keyspace, String indexName)
+    {
+        for (int nodeId = 1; nodeId <= cluster.size(); nodeId++)
+        {
+            for (int replica = 1; replica <= cluster.size(); replica++)
+            {
+                NodeIndex nodeIndex = NodeIndex.create(keyspace, indexName, cluster.get(replica));
+                Index.Status expected = expectedNodeIndexQueryability.get(nodeIndex);
+
+                assertIndexingStatus(cluster.get(nodeId), keyspace, indexName, cluster.get(replica), expected);
+            }
+        }
+    }
+
+    private static void assertIndexingStatus(IInvokableInstance node, String keyspaceName, String indexName, IInvokableInstance replica, Index.Status expected)
+    {
+        InetAddressAndPort replicaAddressAndPort = getFullAddress(replica);
+        try
+        {
+            Index.Status actual = getNodeIndexStatus(node, keyspaceName, indexName, replicaAddressAndPort);
+            String errorMessage = String.format("Failed to verify %s.%s status for replica %s on node %s, expected %s, but got %s.",
+                                                keyspaceName, indexName, replica.broadcastAddress(), node.broadcastAddress(), expected, actual);
+            assertEquals(errorMessage, expected, actual);
+        }
+        catch (Exception e)
+        {
+            throw new RuntimeException(e);
+        }
+    }
+
+    private static void waitForIndexingStatus(IInvokableInstance node, String keyspace, String index, IInvokableInstance replica, Index.Status status)
+    {
+        InetAddressAndPort replicaAddressAndPort = getFullAddress(replica);
+        await().atMost(5, TimeUnit.SECONDS)
+               .until(() -> node.callOnInstance(() -> getIndexStatus(keyspace, index, replicaAddressAndPort) == status).booleanValue());
+    }
+
+    private static Index.Status getNodeIndexStatus(IInvokableInstance node, String keyspaceName, String indexName, InetAddressAndPort replica)
+    {
+        return Index.Status.values()[node.callsOnInstance(() -> getIndexStatus(keyspaceName, indexName, replica).ordinal()).call()];
+    }
+    
+    private static Index.Status getIndexStatus(String keyspaceName, String indexName, InetAddressAndPort replica)
+    {
+        KeyspaceMetadata keyspace = Schema.instance.getKeyspaceMetadata(keyspaceName);
+        if (keyspace == null)
+            return Index.Status.UNKNOWN;
+
+        TableMetadata table = keyspace.findIndexedTable(indexName).orElse(null);
+        if (table == null)
+            return Index.Status.UNKNOWN;
+
+        SecondaryIndexManager indexManager = Keyspace.openAndGetStore(table).indexManager;
+        
+        return indexManager.getIndexStatus(replica, keyspaceName, indexName);
```

Comment:
```suggestion
        return SecondaryIndexManager.getIndexStatus(replica, keyspaceName, indexName);
```

Probably can avoid getting the `indexManager` instance?